bears/general/LineCountBear.py

```diff
@@ -10,14 +12,19 @@ class LineCountBear(LocalBear):
     LICENSE = 'AGPL-3.0'
     CAN_DETECT = {'Formatting'}
 
-    def run(self, filename, file, max_lines_per_file: int):
+    def run(self, filename, file, max_lines_per_file: int,
+            exclude_blank_lines: bool=False):
         """
         Count the number of lines in a file and ensure that they are
         smaller than a given size.
 
         :param max_lines_per_file: Number of lines allowed per file.
+        :param exclude_blank_lines: True if blank lines are to be excluded.
         """
         file_length = len(file)
+        blank_lines = len(list(filter(lambda x: re.match(r'^\s*$', x), file)))
```

Comment:
The compiled RE will definitely be cached in this pipeline (see note in [the docs](https://docs.python.org/2/library/re.html#re.compile), so a method level compiled variable would be useless.
This RE will probably be cached for the entire section being processed. The same bear is run on each file as part of the same stage of processing, and the RE cache probably wont be overflowed if only a few bears are running in a single stage of processing.

Moving this RE to a compiled variable at the module or class level means extra work must be done at startup, which isnt a good idea as this bear may not even be needed by `.coafile`, so the extra work is wasted.